packages/xo-server/src/api/xostor.mjs

```diff
@@ -165,54 +166,62 @@ export const create = defer(async function (
   const task = await this.tasks.create({ name: `creation of XOSTOR: ${name}`, type: 'xo:xostor:create' })
   return task.run(async () => {
     const hostIds = Object.keys(disksByHost)
+    const hosts = hostIds.map(hostId => this.getObject(hostId, 'host'))
+    if (!hosts.every(host => host.$pool === hosts[0].$pool)) {
+      // we need to do this test to ensure it won't create a partial LV group with only the host of the pool of the first master
+      throw new Error('All hosts must be in the same pool')
+    }
 
-    const tmpBoundObjectId = `tmp_${hostIds.join(',')}_${Math.random().toString(32).slice(2)}`
-
-    const license = await Task.run({ properties: { name: 'license check' } }, async () => {
-      const xostorLicenses = await this.getLicenses({ productType: 'xostor' })
+    const host = hosts[0]
+    const xapi = this.getXapi(host)
+    const poolHostIds = Object.keys(xapi.objects.indexes.type.host)
```

Comment:
is it possible to have a host that is not in hostIds , but is in poolHostIds ? Maybe a host without a disk used in this xostore creation ?

Reply:
yes